### PR TITLE
Add: Select Stats Generator

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

Adds a select component that takes a request and lastUpdated time that will generate children function which will provide both the stats that resolve from the request and the error, if any.

## Things to Keep in Mind

1. This component doesn't do any error handling - it's up to the child to figure out what to do with the errors

2. The `selectedTimeRange` state is equal to the option labels because it's a lot harder to keep `Date.now()` in sync with what the actual state is. The labels are static information that isn't changing, so it just made more sense to do it that way.

3. There isn't any loading state - this can obviously be changed, but the Linode and NodeBalancer graphs don't have any loading state, so I figured it might be arbitrary

4. This doesn't do any timezone conversion - this is intentional. The child is responsible for this

## Please read the usage docs

[pls do 😭 ](https://github.com/linode/manager/blob/95978c65e6ed48e37d51a3d47715c35cf6b17397/packages/manager/src/features/Longview/shared/USAGE.md)

## To Test

I created [a quick little gist](https://gist.github.com/martinmckenna/e7db5440a6d06937d2ed83ef59522bf9) for convenience sake. Just replace the entirety of `Installation.tsx` with what's in that gist and navigate to the installation tab.

Please play around with the data, requests and stuff.

## Type of Change
- Non breaking change ('update', 'change')

## In Progress

- [x] Unit Tests

### CODE REVIEW

File: packages/manager/src/containers/profile.container.ts [hunk 1]
```diff
@@ -1,30 +1,78 @@
...
-    (state: ApplicationState, ownProps: TOuter) => {
-      return mapAccountToProps(ownProps, state.__resources.profile);
+  connect<
+    (ReduxState & Partial<StateProps>) | StateProps,
+    DispatchProps,
+    OwnProps,
+    ApplicationState
+  >(
+    (state, ownProps) => {
+      const {
```
Comment (martinmckenna) [inline packages/manager/src/containers/profile.container.ts]: This file change wasn't exactly necessary - but was leftover from an earlier solution I tried.

File: packages/manager/src/features/Longview/shared/TimeRangeSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        .catch(e => {
+          /* always set an error - it's up to the consumer if it wants to do anything with this error */
+          setError(
+            getAPIErrorOrDefault(e, 'There was an issue retrieving the stats')
+          );
+        });
+    }
+  }, [lastUpdated]);
+
+  const handleChange = (item: Item<(now: number) => number, Labels>) => {
```
Comment (Jskobos) [inline packages/manager/src/features/Longview/shared/TimeRangeSelect.tsx]: (Optional:) You could get around this typing by having the value be the same as the label, then calling a helper here to convert the numbers

Comment (martinmckenna) [reply packages/manager/src/features/Longview/shared/TimeRangeSelect.tsx]: this is a better idea

**Comment by johnwcallahan** [comment]
I haven't given this a full review yet, but I have a general question: how will we handle layouts? This component renders the `<Select />` for you, followed by the children. That's not always what we'll want though, like in the design below:

Screen Shot 2019-11-22 at 8 10 18 AM

It seems like we could just expose the props needed for the Select component to the consumer, which could handle the component. It might be better as a hook, for that matter.

**Comment by martinmckenna** [comment]
@johnwcallahan should work fine. The component doesn't nest the children under the Select

```js
      <Select
        {...restOfSelectProps}
        onChange={handleChange}
        isClearable={false}
        isSearchable={false}
        value={options.find(o => o.label === selectedTimeRange) || options[0]}
        options={options}
      />
      {children(stats, fetchError)}
    </React.Fragment>
```

**Comment by johnwcallahan** [comment]
I may be missing something here. How would you create a layout like this?

```tsx
<Grid>
  <Grid>
    <Header />
    <SideBar>
      <Select /> {/* TimeRangeSelect */}
    </SideBar>
  </Grid>
  <Grid>
    <SecondaryHeader />
    <Graph /> {/* Needs props from TimeRangeSelect  */}
  </Grid>
</Grid>
```

The `<Select />` and `<Graph />` are on different levels of nesting. But the `<TimeRangeSelect />` component forces the Select to be at the highest level.

**Comment by martinmckenna** [comment]
@johnwcallahan deleted my comment after re-reading. I'd push back against making it a hook - the idea here was to avoid having to deal with any markup at all.

What might be easier is to just expose a `handleStatsChange` and `handleError` callbacks that get fired in the promise, so that the consumer can decide what it wants to do with those new values.

**Comment by johnwcallahan** [comment]
That seems like it'd work. Then you could do something like:

```tsx
const [stats, setStats] = React.useState();

return (
  <>
    <TimeRangeSelect request={...} handleStatsChange={stats => setStats(stats)} />  
    <Graph stats={stats} />
  </>
);
```

**Comment by martinmckenna** [comment]
@johnwcallahan updated all the code, readme, and testing gist.

**Pull request merged**